### ISSUE TITLE
Entropy loss in continuous actions

### ISSUE THREAD

**Issue by DeepWolf90** [werner-duvaud/muzero-general#175]
In the paper, there is an entropy loss **Eq(17)**  used as policy regularizer to make sure that the policy not converged prematurely to a sub-optimal policy.

**Eq (18)**, the computed entropy loss is added to the overall loss. I believe, the entropy loss needs to be subtracted to encourage for the high entropy diverse policies and hence help in minimizing the overall loss.

Besides, there is no implementation for the entropy loss in the code, so I implemented it as follow:

_(This implementation is generalized to more than one continuous action)_

```
def get_entropy(mu, sigma, action):
    # compute the log_prob of the root sampled actions at each step 
    mu = torch.tanh(mu)
    sigma = torch.exp(sigma)

    m = torch.distributions.normal.Normal(mu, sigma)

    log_prob = m.log_prob(action).sum(1)
    prob = torch.exp(log_prob)
    
    # compute entropy
    entropy = - (prob * log_prob).sum()

    return entropy
```
Then, I subtract it from the overall loss:
`loss = value_loss * self.config.value_loss_weight + reward_loss + policy_loss - (self.config.beta * entropy_loss)
`

**Comment by DeepWolf90** [comment]
Update on computing the entropy loss. The above implementation is actually wrong because the entropy is computed for the sampled actions only. However, we should compute entropy on the entire actions distributions as follow:

```
def get_entropy(mu, sigma):

    mu = torch.tanh(mu)
    sigma = torch.exp(sigma)

    m = torch.distributions.normal.Normal(mu, sigma)
    
    # compute entropy
    entropy = m.entropy().sum()  # sum along actions axis

    return entropy
```

**Comment by DeepWolf90** [comment]
@werner-duvaud 
Is my implementation of entropy loss correct?
and another thing, Do we sum the entropy loss for  each step of the unrolled steps just like we sum the policy and value losses?

            ```
            value_loss += current_value_loss
            reward_loss += current_reward_loss
            policy_loss += current_policy_loss
            entropy_loss += current_entropy_loss

```
loss = (value_loss * self.config.value_loss_weight 
                + reward_loss                                         
                + policy_loss                                                   
                - (self.config.beta * entropy_loss)
                ) 
```

```

**Comment by ahainaut** [comment]
Hello @DeepWolf90,
Thank you for your interest on the project. The continuous version of MuZero has just been updated, including the entropy loss. You can check it out at: [MuZero continous](https://github.com/werner-duvaud/muzero-general/tree/continuous)
